src/components/tl/ucp/alltoallv/alltoallv_onesided.c

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+#include "tl_ucp.h"
+#include "alltoallv.h"
+#include "core/ucc_progress_queue.h"
+#include "utils/ucc_math.h"
+#include "tl_ucp_sendrecv.h"
+
+void ucc_tl_ucp_alltoallv_onesided_progress(ucc_coll_task_t *ctask);
+
+ucc_status_t ucc_tl_ucp_alltoallv_onesided_start(ucc_coll_task_t *ctask)
+{
+    ucc_tl_ucp_task_t *task   = ucc_derived_of(ctask, ucc_tl_ucp_task_t);
+    ucc_tl_ucp_team_t *team   = TASK_TEAM(task);
+    ptrdiff_t          src    = (ptrdiff_t)TASK_ARGS(task).src.info_v.buffer;
+    ptrdiff_t          dest   = (ptrdiff_t)TASK_ARGS(task).dst.info_v.buffer;
+    ucc_rank_t         grank  = UCC_TL_TEAM_RANK(team);
+    ucc_rank_t         gsize  = UCC_TL_TEAM_SIZE(team);
+    long              *pSync  = TASK_ARGS(task).global_work_buffer;
+    ucc_aint_t        *s_disp = TASK_ARGS(task).src.info_v.displacements;
+    ucc_aint_t        *d_disp = TASK_ARGS(task).dst.info_v.displacements;
+    size_t           sdt_size = ucc_dt_size(TASK_ARGS(task).src.info_v.datatype);
+    size_t           rdt_size = ucc_dt_size(TASK_ARGS(task).dst.info_v.datatype);
+    ucc_rank_t       peer;
+    size_t           sd_disp, dd_disp, data_size;
+
+    ucc_tl_ucp_task_reset(task, UCC_INPROGRESS);
+
+    /* perform puts to each member i using that displacement */
+    for (peer = (grank + 1) % gsize; task->onesided.put_posted < gsize;
+         peer = (peer + 1) % gsize) {
+        sd_disp =
+            ucc_coll_args_get_displacement(&TASK_ARGS(task), s_disp, peer) *
+            sdt_size;
+        dd_disp =
+            ucc_coll_args_get_displacement(&TASK_ARGS(task), d_disp, peer) *
+            rdt_size;
+        data_size =
+            ucc_coll_args_get_count(&TASK_ARGS(task),
+                                    TASK_ARGS(task).src.info_v.counts, peer) *
+            sdt_size;
+
+        UCPCHECK_GOTO(ucc_tl_ucp_put_nb((void *)PTR_OFFSET(src, sd_disp),
```

Comment:
it's a detail but: I think that the casting to `void *` is redundant here because it is already done by `PTR_OFFSET`

Reply:
thanks updated